ostd/src/cpu/cpu_local.rs

```diff
@@ -209,13 +223,76 @@ pub(crate) unsafe fn early_init_bsp_local_base() {
 /// It must be guaranteed that the BSP will not access local data before
 /// this function being called, otherwise copying non-constant values
 /// will result in pretty bad undefined behavior.
-pub(crate) unsafe fn init_on_bsp() {
-    // TODO: allocate the pages for application processors and copy the
-    // CPU-local objects to the allocated pages.
+pub unsafe fn init_on_bsp() {
+    let bsp_base_va = __cpu_local_start as usize;
+    let bsp_end_va = __cpu_local_end as usize;
+
+    let nbytes = (bsp_end_va - bsp_base_va).align_up(PAGE_SIZE);
```

Comment:
Move this variable close to `alloc_contiguous`. It is not and should be used by any other code.